instrumentation/github.com/gin-gonic/gin/otelgin/option.go

```diff
@@ -57,3 +64,15 @@ func WithTracerProvider(provider oteltrace.TracerProvider) Option {
 		}
 	})
 }
+
+// WithFilter adds a filter to the list of filters used by the handler.
+// If any filter indicates to exclude a request then the request will not be
+// traced. All filters must allow a request to be traced for a Span to be created.
+// If no filters are provided then all requests are traced.
+// Filters will be invoked for each processed request, it is advised to make them
+// simple and fast.
+func WithFilter(f Filter) Option {
```

Comment:
Instead of requiring the user to specify this multiple times this can be made variadic.

```suggestion
func WithFilter(f ...Filter) Option {
```

The append will need to be updated.